### DISCUSSION TITLE
PARALLEL

### DISCUSSION THREAD

**Discussion by Dewu-Yang** [loganoz/horses3d#162] [Q&A] [answered]
Dear all,

   How can I perform parallel computing with the code? I tried running “mpirun -np ** ./horses3d.ns ***.hsol” but it failed. What is the correct command for parallel execution?

Best,
Dewu

**Comment by loganoz** [comment]
Dear @Dewu-Yang,

Horses3D has openmp and mpi parallelizations. From your command, I understand that you're trying to use mpi. In any case, I might need some extra information.
Where are you running? Is it a desktop or a cluster? How many threads do you plan to use? Have you compiled with the parallel option? 

Best,
Gonzalo

**Reply by Dewu-Yang** [reply]
Dear Gonzalo，
Thank you for your response. I plan to run on a desktop and I intend to use a CPU with 10 cores or more. In the early stages of compilation, I have already selected parallel option (COMM=PARALLEL). Thank you again for your reply and I look forward to your suggestions.

Best,
Dewu

**Reply by Dewu-Yang** [reply, marked as answer]
Dear Gonzalo,

After completing the necessary configuration for dependencies (PETSc, HDF5, METIS, etc.), I have successfully achieved parallel operations. Thanks!

Best,
Dewu
